### PR TITLE
Use fewer queries to fill the "category page with product listing"  

### PR DESCRIPTION
**Pull request by skovsboll**
The number of queries are optimized mainly by removing the http requests to "getproductinformation".
Prices are calculated in one go and rendered server side.

### CODE REVIEW

File: src/Ucommerce.Avenue.Umbraco/Api/AvenueClothingApiProductController.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ public IHttpActionResult GetProductVariations([FromBody] GetProductVariationsReq
                 };
             }).ToList();
 
-            return Json(new { Variations = variations });
+            return Json(new {Variations = variations});
```
[the remark this answers is not in the document]
Comment (vitalie-ucommerce) [reply src/Ucommerce.Avenue.Umbraco/Api/AvenueClothingApiProductController.cs]: what what now? 🙊 
you are right, we need to agree on a formatting standard

File: src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs [hunk 2]
```diff
@@ -17,29 +26,58 @@ public class CategoryController : RenderMvcController
         public ICatalogLibrary CatalogLibrary => ObjectFactory.Instance.Resolve<ICatalogLibrary>();
         public ISiteContext SiteContext => ObjectFactory.Instance.Resolve<ISiteContext>();
         public ISearchLibrary SearchLibrary => ObjectFactory.Instance.Resolve<ISearchLibrary>();
+        public IRepository<Currency> CurrencyRepository => ObjectFactory.Instance.Resolve<IRepository<Currency>>();
```
Comment (vitalie-ucommerce) [inline src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs]: this one is not used, no?

File: src/Ucommerce.Avenue.Umbraco/Controllers/Constants.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace Ucommerce.Avenue.Umbraco.Controllers
+{
+    public class Constants
+    {
+        public static readonly Guid CatalogGuid = new Guid("55BEFD9E-EFE8-6475-9BFC-C2A37349C708");
```
Comment (vitalie-ucommerce) [inline src/Ucommerce.Avenue.Umbraco/Controllers/Constants.cs]: Can I ask why we have this hardcoded?

File: src/Ucommerce.Avenue.Umbraco/Api/AvenueClothingApiProductController.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (skovsboll) [reply src/Ucommerce.Avenue.Umbraco/Api/AvenueClothingApiProductController.cs]: I've set up Rider to ignore extra s p a c e s when formatting ;)

File: src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs (hunk 2 above)
Comment (skovsboll) [reply src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs]: You're right. Thanks!

File: src/Ucommerce.Avenue.Umbraco/Controllers/Constants.cs (hunk 3 above)
Comment (skovsboll) [reply src/Ucommerce.Avenue.Umbraco/Controllers/Constants.cs]: Certainly. In the case where we have two active ProductCatalogs, it was picking one indeterministically. I wanted to fixate it.
However, I find it much easier to just uncheck "Display On Site" for all but one and then run the indexing.
Thanks for raising the question.

**Pull request merged**